File: scratch/extract_docx_tags.dart
import 'dart:io';
import 'package:archive/archive.dart';

void main() {
  final bytes = File('e:/Codes/kasudlo/FINAL-CDX-TOOL-2024-1_TAGGED.docx').readAsBytesSync();
  final archive = ZipDecoder().decodeBytes(bytes);
  
  final docXml = archive.findFile('word/document.xml');
  if (docXml == null) {
    print('document.xml not found');
    return;
  }
  
  final content = String.fromCharCodes(docXml.content as List<int>);
  
  // Strip XML tags
  String text = content.replaceAll(RegExp(r'<[^>]+>'), '');
  
  // Unescape XML entities
  text = text.replaceAll('&lt;', '<').replaceAll('&gt;', '>').replaceAll('&amp;', '&');
  
  // Find Docmosis tags
  final tags = RegExp(r'<<.*?>>').allMatches(text).map((m) => m.group(0)!).toSet().toList()..sort();
  
  for (final tag in tags) {
    print(tag);
  }
}

File: scratch/extract_docx_tags.py
import zipfile
import re
import sys

def extract_tags(docx_path):
    with zipfile.ZipFile(docx_path) as docx:
        content = docx.read('word/document.xml').decode('utf-8')
    
    # Remove XML tags to get pure text
    text = re.sub(r'<[^>]+>', '', content)
    
    # Find all << tags >>
    tags = re.findall(r'<<.*?>>', text)
    
    # Print unique tags
    for tag in sorted(set(tags)):
        print(tag)

if __name__ == "__main__":
    extract_tags(sys.argv[1])
